test/suites/integration/ami_test.go

```diff
@@ -211,6 +213,34 @@ var _ = Describe("AMI", func() {
 
 			env.ExpectInstance(pod.Spec.NodeName).To(HaveField("ImageId", HaveValue(Equal(customAMI))))
 		})
+		It("should have the AWSNodeTemplateStatus for AMIs using wildcard", func() {
+			provider := awstest.AWSNodeTemplate(v1alpha1.AWSNodeTemplateSpec{
+				AWS: v1alpha1.AWS{
+					SecurityGroupSelector: map[string]string{"karpenter.sh/discovery": settings.FromContext(env.Context).ClusterName},
+					SubnetSelector:        map[string]string{"karpenter.sh/discovery": settings.FromContext(env.Context).ClusterName},
+				},
+				AMISelector: map[string]string{"aws::name": "*"},
+			})
+
+			env.ExpectCreated(provider)
+			ant := EventuallyExpectAMIsToExist(provider)
+			Expect(len(ant.Status.AMIs)).To(BeNumerically("<", 10))
```

Comment:
Is it possible for us to know exactly how many we will have?

Reply:
Should this be `<=25` as we're limiting it to 25 now?

Reply:
No, we are not limiting AMIs to be less than 25 in the status. I believe you are referring to the what will be displayed in the logs.